torchmultimodal/modules/layers/transformer.py

```diff
@@ -328,11 +334,68 @@ def forward(
             )
 
 
-def _apply_layernorm(x: Tensor, layernorm: nn.Module) -> Tensor:
-    """Supports mixed-precision training by casting to fp32 for layernorm and back"""
-    if x.dtype != torch.float32:
-        x_fp32 = x.float()
-        x_fp32 = layernorm(x_fp32)
-        return x_fp32.type_as(x)
-    else:
-        return layernorm(x)
+class TransformerEncoder(nn.Module):
+    def __init__(
+        self,
+        n_layer: int,
+        d_model: int,
+        n_head: int,
+        dim_feedforward: int,
+        dropout: float = 0.0,
+        activation: Callable[..., nn.Module] = nn.ReLU,
+        layer_norm_eps: float = 1e-12,
+        norm_first: bool = False,
+    ):
+        super().__init__()
+        self.layer = nn.ModuleList(
+            [
+                TransformerEncoderLayer(
+                    d_model,
+                    n_head,
+                    dim_feedforward,
+                    dropout,
+                    activation,
+                    layer_norm_eps,
+                    norm_first,
+                )
+                for _ in range(n_layer)
+            ]
+        )
+        self.num_layers = n_layer
```

Comment:
not used?

Reply:
hmmm I think it was used in [CLIPTextEncoder](https://github.com/facebookresearch/multimodal/blob/f9327dec04a5e8c824fba36e4ed52528e44541f6/torchmultimodal/modules/encoders/clip_text_encoder.py#L80). I won't be able to get to unifying that component but maybe we should keep `self.num_layers` for when we do?

Reply:
i think the caller can maintain the num layers. no need to add unused instance variables here

Reply:
aight I'll remove it